common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,31 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
-    } else {
-      return text;
+
+    if (isQuoted(text, "\"")) {
+      return text.substring(1, text.length() - 1)
+          .replace("\\\"", "\"")
```

Comment:
Please consider using something like replaceAll instead of trying to cover each of these cases individually.  We shouldn't need to consider both `\\` and `\\\\` and `\\\\\\` and ...etc